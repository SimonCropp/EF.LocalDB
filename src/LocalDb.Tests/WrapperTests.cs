using System;
using System.Data.SqlClient;
using System.Threading.Tasks;
using ApprovalTests;
using Xunit;
using Xunit.Abstractions;

public class WrapperTests :
    XunitLoggingBase
{
    static Wrapper instance;

    [Fact]
    public void InvalidInstanceName()
    {
        var exception = Assert.Throws<ArgumentException>(() => new Wrapper("<", "s", 4));
        Approvals.Verify(exception.Message);
    }

    [Fact]
    public async Task Delete()
    {
        var (connection, id) = await instance.CreateDatabaseFromTemplate("ToDelete");
        await instance.DeleteDatabase("ToDelete");

        using (var sqlConnection = new SqlConnection(instance.MasterConnectionString))
        {
            await sqlConnection.OpenAsync();
            var settings = DbPropertyReader.Read(sqlConnection, id);
            Assert.Empty(settings.Files);
        }
    }

    public WrapperTests(ITestOutputHelper output) :
        base(output)
    {
    }

    static WrapperTests()
    {
        instance = new Wrapper("WrapperTests", DirectoryFinder.Find("WrapperTests"), 4);
        instance.Start(DateTime.Now, TestDbBuilder.CreateTable);
    }
}