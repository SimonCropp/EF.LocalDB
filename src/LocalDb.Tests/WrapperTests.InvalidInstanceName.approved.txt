Invalid file name: <
Parameter name: instance